sencha-workspace/SlateCSV/app/view/ImporterController.js

```diff
@@ -257,5 +264,69 @@ Ext.define('SlateCSV.view.ImporterController', {
         } else {
             previewGrid.setHidden(true);
         }
+    },
+
+    onImportButtonClick: function() {
+        var me = this,
+            view = me.getView(),
+            useFirstRowForColumnNames = view.getUseFirstRowForColumnNames(),
+            comboBoxes = me.getValidComboBoxes(),
+            csvData = view.getCsvData(),
+            cols = Ext.Object.getKeys(csvData[0]),
+            vtypes = Ext.form.VTypes,
+            validations = [],
+            validationWindow = view.getValidationWindow(),
+            failures = 0;
+
+        Ext.Array.each(comboBoxes, function(combo) {
+            field = combo.findRecordByValue(combo.getValue());
+
+            if (field.get('vtype') && vtypes[field.get('vtype')] ) {
+                validations.push({
+                    fieldName: field.get('fieldName'),
+                    fieldLabel: field.get('label'),
+                    dataIndex: combo.dataIndex,
+                    label: combo.getFieldLabel(),
+                    vtype: field.get('vtype'),
+                    vtext: (field.get('vtypeText') || vtypes[field.get('vtype') +'Text']),
+                    invalidRows: []
+                });
+            }
+        });
+
+        Ext.Array.each(csvData, function(row,index) {
+            Ext.Array.each(validations, function(v) {
```

Comment:
@BillClinton please review these documents as soon as you get a chance and adhere to them as best you can going forward: https://github.com/sencha/code-guidelines

Specifically regarding loops:
- https://github.com/sencha/code-guidelines/blob/master/JavaScript_Performance.md#loops
- https://github.com/sencha/code-guidelines/blob/master/JavaScript_Performance.md#function-based-iteration